### ISSUE TITLE
Add a settings option to turn formatter on and off

### ISSUE THREAD

**Issue by witnessmonolith** [godotengine/godot-vscode-plugin#580]
### Godot version

4/irrelevant

### VS Code version

irrelevant

### Godot Tools VS Code extension version

2.0.0

### System information

irrelevant

### Problem statement

It would be nice to have an on/off option for the built-in code formatter, because, as of right now, it is still in development and would introduce problems and inconsistencies from time to time, and other parts of the extension shouldn't be affected by this.

My point is, disabling the formatter will unblock users who don't care as much for the code formatting, those, who are using alternatives such as [Scony's](https://github.com/Scony/godot-gdscript-toolkit) solution, and those, who are willing to use and test some other functionality of the extension aside from the formatting.

I tried to play around the `settings.json` but to no luck:

```json
{
	"editor.defaultFormatter": null,
	"[gdscript]": {
		"editor.defaultFormatter": null, // it's set to "geequlim.godot-tools" automatically somewhere?
	}
}
```

Also, didn't find the option to turn the formatter off (if it's already exists).

### Proposed solution

To have something like 

```json
"godotTools.editor.enableFormatter": true,
```

available for the configuration.

**Comment by DaelonSuzuka** [comment]
Formatters in VSCode are only invoked when you request them, so I don't understand why this would be necessary. Do you have format-on-save enabled?

Maybe you could try:

```json
{
	"[gdscript]": {
		"editor.formatOnSave": false
	}
}
```

**Comment by microaeris** [comment]
@DaelonSuzuka Thanks for the suggestion. I tried this:
```json
    "editor.formatOnSave": true,
    "[gdscript]": {
        "editor.formatOnSave": false
    },
```
but doesn't work. Looks like "gdscript" [isn't a known language identifier by VSCode](https://code.visualstudio.com/docs/languages/identifiers).

**Comment by DaelonSuzuka** [comment]
When are your files actually being formatted? Do you have `editor.formatOnType` enabled too (why would anybody want format-on-type?)? For troubleshooting purposes, what happens if you disable formatOnSave entirely>

> Does the extension define the language identifier?

Yes, `gdscript` is the identifier for the GDScript language, and it's defined by this extension.

I have this block:

```json
    "[python]": {
        "editor.rulers": [
            120,
            150
        ],
        "editor.defaultFormatter": "charliermarsh.ruff",
        "editor.formatOnType": true
    },
```

in my VSCode settings, so I'm pretty sure that's the correct syntax for per-language settings.

**Comment by microaeris** [comment]
Thank you! I figured out my issue. Tl;dr: my test was bad. and this does work:
```
    "editor.formatOnSave": true,
    "[gdscript]": {
        "editor.formatOnSave": false
    },
```

Thanks for your help!

**Comment by DaelonSuzuka** [comment]
Awesome, glad I could help.
